Fixes to HTTP compression behavior

dgilman opened the pull request:
- Fix build on old MacOS X
- Leave compression OFF by default
- For livecheck, turn it ON by default
- Expose livecheck compression toggle to portfile via livecheck.compression

tobypeterson commented on src/pextlib1.0/curl.c:
```diff
@@ -283,8 +290,8 @@ CurlFetchCmd(Tcl_Interp* interp, int objc, Tcl_Obj* CONST objv[])
 				}
-         } else if (strcmp(theOption, "--disable-compression") == 0) {
-				acceptEncoding = NULL;
+         } else if (strcmp(theOption, "--enable-compression") == 0) {
```
This line is indented with spaces instead of tabs.

dgilman commented:
I've fixed up the indentation across the entire file and added a "modeline".  It is the vim settings I have to get the thing to indent according to the style used in the rest of the file but I am having trouble in my personal vim setup getting it to properly read the modeline on start.  If there are any vim wizards here I'd appreciate feedback, but even if the modeline is non functional it at least serves as documentation of the correct formatting.

raimue commented:
I think the modeline and whitespace re-indentation is not related to this feature and I do not see the necessity to make these changes across the whole file. Let's discuss this separately. Could you please move this change to a new pull request instead?

jmroot commented on src/pextlib1.0/curl.c:
```diff
@@ -1,5 +1,6 @@
 /*
  * curl.c
+ * vim: set noet ts=8 sw=8 tw=80 sts=-1 cino=t0 :
```
If you add a modeline, please use the standard MacPorts one.

jmroot commented on src/pextlib1.0/curl.c:
```diff
@@ -59,12 +62,17 @@
 #define _CURL_MINIMUM_PROGRESS_INTERVAL ((double)(0.2)) /* 0.2 seconds */
 
+/* pre-10.8 curl does not appear to support compression out of the box */
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_8
```
This would be better done as a curl version check.

jmroot replied:
Actually, it looks like curl supports compression back to at least 10.6. The use of https in the tests is likely the problem.

jmroot replied:
Hmm, nope, `/usr/bin/curl --compressed https://www.whatsmyip.org/http-compression-test/` works fine, so it's not the TLS. But it isn't lack of CURLOPT_ENCODING either.

dgilman replied:
My hunch was that the Apple curl wasn't compiled with zlib or something like that. If you have an old Mac around can you see what kind of request headers it's producing?  If that is the case it makes sense to just ban on OSX version.

jmroot replied:
```
% /usr/bin/curl -v --compressed https://www.whatsmyip.org/http-compression-test/
* About to connect() to www.whatsmyip.org port 443 (#0)
*   Trying 204.11.35.98... connected
* Connected to www.whatsmyip.org (204.11.35.98) port 443 (#0)
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Server key exchange (12):
* SSLv3, TLS handshake, Server finished (14):
* SSLv3, TLS handshake, Client key exchange (16):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSL connection using DHE-RSA-AES256-SHA
* Server certificate:
* 	 subject: CN=random.whatsmyip.org
* 	 start date: 2019-12-24 19:43:20 GMT
* 	 expire date: 2020-03-23 19:43:20 GMT
* 	 subjectAltName: www.whatsmyip.org matched
* 	 issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
* 	 SSL certificate verify ok.
> GET /http-compression-test/ HTTP/1.1
> User-Agent: curl/7.19.7 (universal-apple-darwin10.0) libcurl/7.19.7 OpenSSL/0.9.8y zlib/1.2.3
> Host: www.whatsmyip.org
> Accept: */*
> Accept-Encoding: deflate, gzip
> 
< HTTP/1.1 200 OK
< Date: Fri, 31 Jan 2020 02:02:58 GMT
< Server: Apache
< X-Powered-By: Web Server Built Entirely of Legos
< Cache-Control: max-age=0
< Expires: Fri, 31 Jan 2020 02:02:58 GMT
< Vary: Accept-Encoding
< Content-Encoding: gzip
< Content-Length: 3177
< Content-Type: text/html; charset=UTF-8
< 
<!DOCTYPE html>
```
(actual page content trimmed)

jmroot replied:
With current master (i.e. none of the changes in this PR applied), running both `curl fetch` and `curl fetch  --disable-compression` in Tcl sends the exact same headers in each case:
```
GET / HTTP/1.1
User-Agent: MacPorts/2.6.99 libcurl/7.19.7
Host: x.x.x.x:1234
Accept: */*
```

jmroot replied:
The problem is that CURLOPT_ENCODING is not defined as a macro, but as part of an enum. Changing the #elif to an #else should fix it.

jmroot commented on src/port1.0/portlivecheck.tcl:
```diff
@@ -56,11 +56,13 @@ default livecheck.name default
 default livecheck.distname default
 default livecheck.version {$version}
 default livecheck.ignore_sslcert no
+default livecheck.compression yes
```
Adding an option for this may be overkill since livecheck will always want decompressed text to search. I guess it doesn't hurt though.

dgilman commented:
The indentation is fixed

dgilman commented:
I've updated the c preprocessor to handle the old definition. Thanks to jmroot for investigating it, I don't have an old mac anywhere.

jmroot commented on src/pextlib1.0/curl.c:
```diff
@@ -44,6 +44,8 @@
 #include <sys/select.h>
 #include <utime.h>
 
+#include <AvailabilityMacros.h>
```
Looks like this is no longer used.

dgilman replied:
Done

jmroot commented:
Thanks!
